### PR TITLE
Serialize queries when queueing, to allow for table based queries to be carried through to queued export

### PR DESCRIPTION
**Pull request by chris-ware**
This allows for an export derived from the table as source using fromTable to pass it's query to the queue.

### CODE REVIEW

**Comment by pxlrbt** [comment]
Thanks for the PR. Shouldn't that already be possible as the query is evaluated before queueing?

**Comment by chris-ware** [comment]
The query is evaluated at point of being queud but it cannot be stored in a job as an object as it contains closures.

**Comment by pxlrbt** [comment]
Yes, I know. But what's the use case you are trying to solve? Where do you want to use the query afterwards?

**Comment by chris-ware** [comment]
The query is utilised as part of the queued export. Without this PR, when you run the following code, the query that is executed is the standard model query, as the queued version of the export does not have access to the livewire component's query:
`ExcelExport::make()->fromTable()->queue()`

With this PR, if you were to set a table to have filters, and use that same code, the same filters would then apply to the queued export.

Please do try and create an resource yourself with a filter that can be applied, apply that filter in both this PR version of the package and then the main branch, and you will see they return different results. 

Basically, it boils down to `fromTable` should work regardless of whether the export is queued or not, and this PR allows it.

**Comment by pxlrbt** [comment]
Thanks for the explanation. Didn't realize the current version doesn't work properly. I'll review and merge it.

**Comment by pxlrbt** [comment]
Still haven't had the time to properly review this. But talked to Bezhan on Discord and he confirmed, that this fixes issues.

**Comment by salvationarinze** [comment]
I get this error Serialization of 'Closure' is not allowed on doing this
```
ExportAction::make()->exports([
                ExcelExport::make()
                    ->fromTable()
                    ->queue()
            ])
```

**Comment by pxlrbt** [comment]
@salvationarinze That's probably because of some method on your tables columns that I didn't serialize yet. Maybe try isolation the column that throws the error and share the code in a discussion or an issue.
